### ISSUE TITLE
Temporal regularization experiment with exchange rate data

### ISSUE THREAD

**Issue by deploy-soon** [deploy-soon/TemporalMF#1]
Some experiment results which contain validation loss among several loss functions.
Exchange rate data is from [Multivariate Time series Data sets](https://github.com/laiguokun/multivariate-time-series-data).

**Exchange Rate**

the collection of the daily exchange rates of eight foreign countries including Australia, British, Canada, Switzerland, China, Japan, New Zealand and Singapore ranging from 1990 to 2016.

shape: (7588, 8)

**Comment by deploy-soon** [comment]
Base Matrix Factorization
- code: [`983ff9`](https://github.com/deploy-soon/TemporalMF/commit/983ff91356a034e96fbcf4864b367043379e6427?diff=unified#diff-7fb0ca18c7a29df43679b7a368e7e3b8)
- loss function
![image](https://user-images.githubusercontent.com/44163510/75543102-71298f80-5a64-11ea-8e07-8c311f1d8b3f.png)
- Adam optimizer(learning rate=0.005)
- train ratio: 0.8
- metric:
![image](https://user-images.githubusercontent.com/44163510/75543454-29573800-5a65-11ea-81f1-fe1e96da7eaf.png)
- result

|**Factors** | 20     | 30     | 40     | 50     |
|---------|--------|--------|--------|--------|
| **NRMSE**   | 0.1625 | 0.1562 | *0.1515 | 0.1525 |

**Comment by deploy-soon** [comment]
L2 Reg Matrix Factorization
- code: [`983ff9`](https://github.com/deploy-soon/TemporalMF/commit/983ff91356a034e96fbcf4864b367043379e6427?diff=unified#diff-7fb0ca18c7a29df43679b7a368e7e3b8)
- loss function
![image](https://user-images.githubusercontent.com/44163510/75551705-1ef26980-5a78-11ea-9d07-3e4c9404aab9.png)
- Adam optimizer(learning rate=0.005)
- train ratio: 0.8
- metric:
![image](https://user-images.githubusercontent.com/44163510/75543454-29573800-5a65-11ea-81f1-fe1e96da7eaf.png)
- result

| Factors      | 20     | 30     | 40     | 50 |
|--------------|--------|--------|--------|----|
| lambda=5.0   | 1.212  | 1.212  | 1.212  |    |
| lambda=0.5   | 0.8070 | 0.8069 | 0.8070 |    |
| lambda=0.05  | 0.1992 | 0.2065 | 0.2026 |    |
| lambda=0.005 | *0.1496 | 0.1497 | 0.1500 |    |

When lamda is 5.0, it seems all prediction set to zero because of over regularized

**Comment by deploy-soon** [comment]
Temporal L2 Reg Matrix Factorization
- code: [`10d386`](https://github.com/deploy-soon/TemporalMF/commit/10d386605e47fb931e67027e7b48c5058d2ac242)
- loss function
![image](https://user-images.githubusercontent.com/44163510/75846026-4d789780-5e1e-11ea-8cf7-fb51e509ffa6.png)
- Adam optimizer(learning rate=0.005)
- train ratio: 0.8
- metric:
![image](https://user-images.githubusercontent.com/44163510/75543454-29573800-5a65-11ea-81f1-fe1e96da7eaf.png)
- lag set: [1, 2, 3, 4, 5, 6]


**result**
- temporal lambda = lambda / 10.0

| Factors      | 20     | 30     | 40     | 50 |
|--------------|--------|--------|--------|----|
| lambda=5.0   | 1.217  | 1.215  | 1.212  |  1.209  |
| lambda=0.5   | 0.6011 | 0.5900 | 0.5947 |  0.5931  |
| lambda=0.05  | 0.1772 | 0.1759 | 0.1805 |  0.1794  |
| lambda=0.005 | 0.15216  | 0.1497 | 0.1484 |  0.1504  |

  - temporal lambda = lambda / 1.0

| Factors      | 20     | 30     | 40     | 50 |
|--------------|--------|--------|--------|----|
| lambda=5.0   | 1.212  | 1.215  | 1.210  | 1.212   |
| lambda=0.5   | 0.5674 | 0.5623 | 0.5644 |  0.5664  |
| lambda=0.05  | 0.1380 | 0.1421 | 0.1413 |  0.1383  |
| lambda=0.005 | 0.09103 | 0.08628 | 0.09126 |  0.08682  |


  - temporal lambda = lambda / 0.1

| Factors      | 20     | 30     | 40     | 50 |
|--------------|--------|--------|--------|----|
| lambda=5.0   | 1.211  |  1.211 | 1.213  |  1.212  |
| lambda=0.5   | 0.5238 | 0.5237 | 0.5123 |  0.5144  |
| lambda=0.05  | 0.09895 | 0.0990 | 0.1010 | 0.09841   |
| lambda=0.005 | 0.03695 | 0.03640 | 0.04048 | 0.04122   |


  - temporal lambda = lambda / 0.01

| Factors      | 20     | 30     | 40     | 50 |
|--------------|--------|--------|--------|----|
| lambda=5.0   |  1.215 | 1.212  | 1.213  | 1.209  |
| lambda=0.5   | 0.5137 | 0.5129 | 0.5125 |  0.5128  |
| lambda=0.05  | 0.09215 | 0.09136 | 0.09233 |  0.09328  |
| lambda=0.005 | 0.03506 | 0.03393 | 0.03489 |  0.03500  |
